src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("The index passed is invalid");
+        }
+        if (size == capacity - 1) {
```

Comment:
The condition `size == capacity - 1` might not be correct. You should check if `size == capacity` to determine whether the array needs to grow, as reaching the capacity means there's no room for new elements.